### ISSUE TITLE
Unknown Space listed as a System in trade route jump

### ISSUE THREAD

**Issue by LeakingAmps** [eyeonus/Trade-Dangerous#102]
Noob here, so correct me if I'm wrong, but "Unknown Space" Doesn't seem to be a spot I can jump to, but it's listed in this trade route: 

trade run --fr="LP 931-40/Fernandes Vision" --to="Wyrd/Bokeili Station" --cap=296 --cr=50000000 --ly=12.00 --empty=16.00 --supply=10 --jum=2 --show-jumps --planetary=N --fc=N --od=N --pad=L --hops=20 -vvv --summary --progress

Specifically in this step:

  Load from Tau Ceti/Ortiz Moreno City (45ls, BMk:Y, Pad:L, Plt:N, Flc:N, Ody:N, Shp:Y, Out:Y, Ref:Y):
      296 x Minerals/Bertrandite              16,515cr vs   21,367cr, 23 hrs vs 44 hrs, total:  4,888,440cr
    Jump Tau Ceti, 11.94ly -> Unknown Space, 9.88ly -> Duamta
    Direct: 18.55ly, Trip: 21.82ly
    Expect to gain 1,436,192cr (4,852cr/ton)
    
    Am I missing something here, or is it just a bug?

**Comment by eyeonus** [comment]
Unknown Space means TD has no idea what system the station is in.

The code that does it was introduced here:
https://github.com/eyeonus/Trade-Dangerous/commit/27ab8b397b1276732f1b915cc956792b03bd47ea

Only player-owned stations, that is, fleet carriers, will ever have the system they are in be 'Unknown Space'.

There are two workarounds:
Option 1 is to download the entire systems data, rather than just the populated systems data, so that even when a fleet carrier is in an updated system TD will correctly report the system it is in rather than stating it is in 'Unknown Space' for any carrier in an unpopulated system. Doing this will dramatically increase the size of TD's database, I believe it triples the size, but it is an option in the eddblink import plugin if you want to do that, check the help for what options do what.

Option 2 is to exclude fleet carriers whenever it is possible that your trade starting point, destination, or route passes near the origin point of the map.

(I assumed the map origin is the center of the Galaxy, that is, at or near Sagittarius A*, so I wouldn't expect the problem to come up, like, ever, so imagine my surprise when I see this big report.)

I'll look into finding a different default location for 'Unknown Space' that's better than 0,0,0 if it's a big problem, but there's no way to get rid of thing that causes 'Unknown Space' without having every known system put into the database, which isn't done but default because of space reasons.

tl;dr: import the full system list or do the run with fleet carriers excluded (--fc N)

**Comment by LeakingAmps** [comment]
Option 2 is to exclude fleet carriers whenever it is possible that your trade starting point, destination, or route passes near the origin point of the map.

I do have --fc=N in the run options, is that no longer a functional shortcut for no fleet carriers?

**Comment by eyeonus** [comment]
All I can tell you is that if a station shows up as being in the system 'Unknown Space', that means the system that that station is in, is not a system your instance of TD has in its database.

This is usually caused by fleet carriers, because they can be found in unpopulated systems, and systems with no population are not, by default, imported when importing using the eddblink plugin.

The best advice I can give you is, if you encounter that problem again, do an import, make sure to explicitly import systems, and do the run again.

It is also never a bad idea to do a clean import in case any errors or database corruption have snuck in.

**Comment by LeakingAmps** [comment]
Yeah, if I run into it again, I'll pull a full, clean import :)

**Comment by LeakingAmps** [comment]
Whoops, wrong button ^_^;
